src/Platform/Microsoft.Testing.Platform/Services/ServiceProvider.cs

```diff
@@ -73,7 +73,7 @@ public bool TryAddService(object service)
         return true;
     }
 
-    public IEnumerable<object> GetServicesInternal(
+    public /* for testing purposes */ virtual IEnumerable<object> GetServicesInternal(
```

Comment:
I don't feel super comfortable to allow virtual just for tests.

Reply:
If we don't mark this `virtual` then we cannot make the `ServiceProvider` return a `Mock<FileLoggerProvider>`. This in turn makes testing `CheckTestResultsDirectoryOverrideAndCreateItAsync` more complex to fully cover as well as slower, because we'll have to move a file to a new log folder and the paths will all have to be legit.

Reply:
Reverted, thanks !